### PR TITLE
Fix the coverity defects.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: tests/mocks/Matchers.cpp [hunk 1]
```diff
@@ -60,23 +52,26 @@ bool DataMatcher::InternalMatchAndExplain(
...
+         << (expected == actual ? " == " : " != ")
+         << "0x" << static_cast<int>(actual) << " ("
+         << ((expected >= '!' && expected <= '~') ?
+             static_cast<char>(expected) : ' ')
+         << (expected == actual ? " == " : " != ")
+         << (actual >= '!' && actual <= '~' ? static_cast<char>(actual) : ' ')
+         << ")";
 
-    matched &= expected == actual;
+      matched &= expected == actual;
```
Comment (peternewman) [inline tests/mocks/Matchers.cpp]: Can we have some brackets please? :)

File: tests/mocks/MessageHandlerMock.cpp [hunk 2]
```diff
@@ -49,22 +49,26 @@ bool MessageMatcher::MatchAndExplain(
...
-       << "\n" << std::dec << i << ": 0x" << std::hex
-       << static_cast<int>(expected)
-       << (expected == actual ? " == " : " != ")
-       << "0x" << static_cast<int>(actual) << " ("
-       << ((expected >= '!' && expected <= '~') ? static_cast<char>(expected) :
-           ' ')
-       << (expected == actual ? " == " : " != ")
-       << (actual >= '!' && actual <= '~' ? static_cast<char>(actual) : ' ')
-       << ")";
+      *listener
```
Comment (peternewman) [inline tests/mocks/MessageHandlerMock.cpp]: Can we put this in a TestUtils file somewhere rather than duplicating it across two files?

**Comment by peternewman** [comment]
Looks good generally.

File: tests/mocks/Matchers.cpp (hunk 1 above)
Comment (nomis52) [inline tests/mocks/Matchers.cpp]: Done

File: tests/mocks/MessageHandlerMock.cpp (hunk 2 above)
Comment (nomis52) [inline tests/mocks/MessageHandlerMock.cpp]: I considered it. I want to see how this shakes out a bit more before splitting it into a shared library.

**Pull request merged**